pkg/lib/kitfile/generate.go

```diff
@@ -0,0 +1,274 @@
+// Copyright 2024 The KitOps Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package kitfile
+
+import (
+	"fmt"
+	"io/fs"
+	"kitops/pkg/artifact"
+	"kitops/pkg/output"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/licensecheck"
+)
+
+var modelWeightsSuffixes = []string{
+	".safetensors", ".pkl",
+	// Pytorch suffixes
+	".bin", ".pth", ".pt", ".mar", ".pt2", ".ptl",
+	// Tensorflow
+	".pb", ".ckpt", ".tflite", ".tfrecords",
+	// NumPy
+	".npy", ".npz",
+	// Keras and others
+	".keras", ".h5", ".caffemodel", ".pmml", ".coreml",
+	// Other suffixes
+	".gguf", ".ggml", ".ggmf", ".llamafile", ".onnx",
+}
+
+var docsSuffixes = []string{
+	".md", ".adoc", ".html", ".pdf",
+}
+
+var metadataSuffixes = []string{
+	".json", ".yaml", ".xml", ".csv", ".txt",
+}
+
+var datasetSuffixes = []string{
```

Comment:
How about `.parquet` and `.csv`?

Reply:
I'm including `.csv` in metadata, though maybe it makes more sense as a dataset exclusively.

Reply:
Added.